### PR TITLE
gg: Added polygon-related functions

### PR DESCRIPTION
**Pull request by lucastarche**
Added two new functions to [gg.v](https://github.com/vlang/v/blob/master/vlib/gg/gg.v): draw_convex_poly() and draw_empty_poly()
The first is implemented by drawing a triangle fan, while the second is implemented with a line strip.

draw_convex_poly:
```v
//Draws a convex polygon, given an array of x coordinates and an array of y coordinates, and a color.
//Note that the points must be given in clockwise order.
pub fn (ctx &Context) draw_convex_poly(x []f32, y []f32, c gx.Color) {
	assert x.len == y.len
	len := x.len
	assert len >= 3

	if c.a != 255 {
		sgl.load_pipeline(ctx.timage_pip)
	}
	sgl.c4b(c.r, c.g, c.b, c.a)

	sgl.begin_triangle_strip()
	for i in 0 .. (len / 2) {
		sgl.v2f(x[0], y[0])
		sgl.v2f(x[i * 2 + 1], y[i * 2 + 1])
		sgl.v2f(x[i * 2 + 2], y[i * 2 + 2])
	}

	if len % 2 == 0 {
		sgl.v2f(x[len - 2], y[len - 2])
	}
	sgl.end()
}
```

draw_empty_poly:
```v
//Draws the borders of a polygon, given an array of x coordinates and an array of y coordinates, and a color.
//Note that the points must be given in clockwise order.
pub fn (ctx &Context) draw_empty_poly(x []f32, y []f32, c gx.Color) {
	assert x.len == y.len
	assert x.len >= 3

	if c.a != 255 {
		sgl.load_pipeline(ctx.timage_pip)
	}
	sgl.c4b(c.r, c.g, c.b, c.a)

	sgl.begin_line_strip()
	for i in 0 .. (x.len) {
		sgl.v2f(x[i], y[i])
	}
	sgl.v2f(x[0], y[0])
	sgl.end()
}
```

Note: given that since there isn't an easy way of testing graphics libraries, I added a new example (examples/gg/polygons.v) instead.

### CODE REVIEW

File: vlib/gg/gg.v [hunk 1]
```diff
@@ -322,16 +322,6 @@ pub fn (ctx &Context) draw_rect(x f32, y f32, w f32, h f32, c gx.Color) {
...
-[inline]
-pub fn (ctx &Context) draw_square(x f32, y f32, s f32, c gx.Color) {
-	ctx.draw_rect(x, y, s, s, c)
-}
-
-[inline]
-pub fn (ctx &Context) set_pixel(x f32, y f32, c gx.Color) {
-	ctx.draw_square(x, y, 1, c)
-}
-
```
Comment (spytheman) [inline vlib/gg/gg.v:334]: why were these functions deleted?

File: vlib/gg/gg.v [hunk 2]
```diff
@@ -366,11 +356,6 @@ pub fn (ctx &Context) draw_empty_rect(x f32, y f32, w f32, h f32, c gx.Color) {
 	sgl.end()
 }
 
-[inline]
-pub fn (ctx &Context) draw_empty_square(x f32, y f32, s f32, c gx.Color) {
-	ctx.draw_empty_rect(x, y, s, s, c)
-}
-
 pub fn (ctx &Context) draw_circle_line(x f32, y f32, r int, segments int, c gx.Color) {
```
Comment (spytheman) [inline vlib/gg/gg.v:359]: same - why was this deleted?

**Comment by spytheman** [comment]
I think the deleted functions were an accident, I am going to restore them and run vfmt on this PR.
Otherwise, good work.

**Comment by spytheman** [comment]
hm, is there a reason that the points are passed as 2 separate arrays x and y, and not as one array of `[x1,y1, x2,y2, x3,y3 ...]`?
In the second case, we would have just `pub fn (ctx &Context) draw_convex_poly(points []f32, c gx.Color) {` ?

**Comment by lucastarche** [comment]
Hello, the deleted function were, in fact a mistake from my part haha.
The reason why I am passing two arrays instead of one was to keep it consistent with the surrounding functions (draw_rect(), draw_triangle(), etc.), which all took the x and y values as separate parameters, but I wouldn't mind passing an array of points instead.

**Comment by medvednikov** [comment]
An array of points is better.

**Comment by lucastarche** [comment]
Changed function signatures to be arrays of points instead.

**Comment by xplshn** [comment]
Excellent, I approve.
